packages/core/src/services/providerApiKeys/create.ts

```diff
@@ -17,17 +17,22 @@ export type Props = {
   token: string
   url?: string
   name: string
+  defaultModel?: string
   author: User
 }
 export function createProviderApiKey(
-  { workspace, provider, token, url, name, author }: Props,
+  { workspace, provider, token, url, name, defaultModel, author }: Props,
   db = database,
 ) {
   return Transaction.call(async (tx) => {
     if (provider === Providers.Custom && !url) {
       return Result.error(new BadRequestError('Custom provider requires a URL'))
     }
 
+    if (defaultModel === '') {
```

Comment:
The default behaviour of the uncontrolled model input, of the create provider modal, was sending the model as an empty string when the input is empty. However, I also controlled this in the modal:

```typescript
onSubmit: (payload) => ({
      ...payload,
      defaultModel: payload.defaultModel || undefined, // Ensure defaultModel is either non-empty or undefined
    }),
````

I think it would be nice to let this check there to ensure no default model is saved as an empty string, because then we would have to check in all places not to use it as the default model if it is an empty string and not "null".